packages/components/src/components/hds/table/index.ts

```diff
@@ -0,0 +1,313 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { assert } from '@ember/debug';
+import { tracked } from '@glimmer/tracking';
+import type { ComponentLike } from '@glint/template';
+
+import {
+  HdsTableDensityValues,
+  HdsTableThSortOrderValues,
+  HdsTableVerticalAlignmentValues,
+} from './types.ts';
+import type {
+  HdsTableColumn,
+  HdsTableDensities,
+  HdsTableHorizontalAlignment,
+  HdsTableOnSelectionChangeArgs,
+  HdsTableSelectableRow,
+  HdsTableSortingFunction,
+  HdsTableThSortOrder,
+  HdsTableVerticalAlignment,
+} from './types';
+import type { HdsFormCheckboxBaseSignature } from '../form/checkbox/base';
+import type { HdsTableTdArgs } from './td.ts';
+import type { HdsTableThArgs } from './th.ts';
+import type { HdsTableThSortArgs } from './th-sort.ts';
+import type { HdsTableTrArgs } from './tr.ts';
+
+export const DENSITIES: HdsTableDensities[] = Object.values(
+  HdsTableDensityValues
+);
+export const DEFAULT_DENSITY = HdsTableDensityValues.Medium;
+
+export const VALIGNMENTS: HdsTableVerticalAlignment[] = Object.values(
+  HdsTableVerticalAlignmentValues
+);
+export const DEFAULT_VALIGN = HdsTableVerticalAlignmentValues.Top;
+
+export interface HdsTableArgs {
+  Args: {
+    align?: HdsTableHorizontalAlignment;
+    caption?: string;
+    columns?: HdsTableColumn[];
+    density?: HdsTableDensities;
+    identityKey?: string;
+    isFixedLayout?: boolean;
+    isSelectable?: boolean;
+    isStriped?: boolean;
+    model: Array<Record<string, unknown>>;
+    onSelectionChange?: (selection: HdsTableOnSelectionChangeArgs) => void;
+    onSort?: (sortBy: string, sortOrder: HdsTableThSortOrder) => void;
+    selectionAriaLabelSuffix?: string;
+    sortBy?: string;
+    sortedMessageText?: string;
+    sortOrder?: HdsTableThSortOrder;
+    valign?: HdsTableVerticalAlignment;
+  };
+  Blocks: {
+    head?: [
+      {
+        Tr: ComponentLike<HdsTableTrArgs>;
+        Th: ComponentLike<HdsTableThArgs>;
+        ThSort: ComponentLike<HdsTableThSortArgs>;
+        sortBy?: string;
+        sortOrder?: HdsTableThSortOrder;
+        setSortBy: (column: string) => void;
+      },
+    ];
+    body?: [
+      {
+        Td: ComponentLike<HdsTableTdArgs>;
+        Tr: ComponentLike<HdsTableTrArgs>;
+        Th: ComponentLike<HdsTableThArgs>;
+        data?: Array<Record<string, unknown>>;
```

Comment:
question: checking my understanding here, isn't `data` a single record, instead of an array?

Reply:
yes, fixing now.